core/domain/collection_domain.py

```diff
@@ -1075,6 +1075,14 @@ def add_skill(self, skill_name):
         self.skills[skill_id] = CollectionSkill(skill_id, skill_name, [])
         self.next_skill_id += 1
 
+    def update_skill(self, skill_id, new_skill_name):
+        """Renames skill with specified id to the new skill name."""
+        if skill_id not in self.skills:
+            raise ValueError(
+                'Skill with ID "%s" does not exist.' % skill_id)
+
+        self.skills[skill_id].name = new_skill_name
```

Comment:
On page 3 of the technical doc, it looks like there's a requirement that skill names in a collection be unique. Can we check that this applies (i.e. the new_skill_name is not already used) before setting it here? There should be tests that verify this, too.